esmvaltool/diag_scripts/shared/_validation.py

```diff
@@ -82,22 +46,26 @@ def get_control_exper_obs(short_name, input_data, cfg, cmip_type):
         logger.info("Observations dataset(s) %s",
                     [obs['dataset'] for obs in obs_selection])
 
-    # determine CONTROL and EXPERIMENT datasets
+    # make sure the chosen datasets for control and exper are available
+    alias_selection = [model['alias'] for model in dataset_selection]
+
+    if cfg['control_model'] not in alias_selection:
+        logger.error("Control dataset %s not in datasets",
+                     cfg['control_model'])
```

Comment:
```suggestion
```

Reply:
no need to both print out a logger error *and* raise an exception, raising an exception is enough